lib/fizzy/parser.hpp

```diff
@@ -79,7 +79,11 @@ inline parser_result<std::vector<T>> parse_vec(const uint8_t* pos, const uint8_t
     std::tie(size, pos) = leb128u_decode<uint32_t>(pos, end);
 
     std::vector<T> result;
-    result.reserve(size);
+
+    // Reserve memory for vec elements if `size` value is reasonable.
+    if (size < 128)
```

Comment:
I think a large number would be still reasonable, especially for the `Data` section. For everything else probably 256 would be more realistic.

Reply:
1. `Data` parsing does not use `parse_vec()` any more.
2. In all benchmarks max size is 2.
3. This still works correctly in else case.
4. I used `128` because `cmp` instruction is smaller then.

Reply:
> Data parsing does not use parse_vec() any more.

But the `Data` section itself is parsed into a vector.

> In all benchmarks max size is 2.

Yes, our benchmarks are tiny. I had another benchmark in the pipeline from ewasm-precompiles which is quite big, that should help us see more realistic numbers.

> I used 128 because cmp instruction is smaller then.

That's nice, but does it have a performance difference? In any case most of this conversation is pretty useless unless we benchmark all these options on more realistic data. **I suggest to keep a note to revisit this in the future.**